serde/src/tagger.rs

```diff
@@ -0,0 +1,23 @@
+/// Provides the correct tag for a given format.
+pub trait Tagger {
```

Comment:
Right now you have a method for most non aggregates. What I'm wondering if it would work to pass something like a `TagDeserializer` to `visit_tagged_value` and then use the regular deserialization framework to deserialize to your target type.

So [this function](https://github.com/pyfisch/cbor/commit/96f67b5c01e264c749d4df4ff25e1005dcce87c7#diff-dd26c2766e65aca43e4228e64ebd0cf8R312) would look like that:

```rust
fn visit_tagged_value<T, D>(&mut self, tag: T, deserializer: &mut D) -> Result<Self::Value, D::Error>
                    where T: de::Deserializer, D: de::Deserializer 
{
    let val = try!(de::Deserialize::deserialize(deserializer));
    let tag = try!(de::Deserialize::deserialize(tagger));
    Ok(Value::Tag(tag, Box::new(val)))
}
```

I haven't looked at serialization closely enough yet.